Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java

```diff
@@ -463,74 +466,52 @@ public static boolean displayAlertMsg(final String caption, final String text,
     }
     else
     {
-      // Create object used for waiting.
-      final Object lock = new Object();
-      AlertDialog.Builder builder = new AlertDialog.Builder(emulationActivity,
-              R.style.DolphinDialogBase)
-              .setTitle(caption)
-              .setMessage(text);
-
-      // If not yes/no dialog just have one button that dismisses modal,
-      // otherwise have a yes and no button that sets alertResult accordingly.
-      if (!yesNo)
-      {
-        builder
-                .setCancelable(false)
-                .setPositiveButton("OK", (dialog, whichButton) ->
-                {
-                  dialog.dismiss();
-                  synchronized (lock)
-                  {
-                    lock.notify();
-                  }
-                });
-      }
-      else
-      {
-        alertResult = false;
-
-        builder
-                .setPositiveButton("Yes", (dialog, whichButton) ->
-                {
-                  alertResult = true;
-                  dialog.dismiss();
-                  synchronized (lock)
-                  {
-                    lock.notify();
-                  }
-                })
-                .setNegativeButton("No", (dialog, whichButton) ->
-                {
-                  alertResult = false;
-                  dialog.dismiss();
-                  synchronized (lock)
-                  {
-                    lock.notify();
-                  }
-                });
-      }
+      sIsAlertMessage = true;
+      sAlertMessageText = text;
 
-      // Show the AlertDialog on the main thread.
-      emulationActivity.runOnUiThread(builder::show);
+      emulationActivity.runOnUiThread(() -> AlertMessage.newInstance(caption, text, yesNo)
+              .show(emulationActivity.getSupportFragmentManager(), "PanicAlert"));
 
       // Wait for the lock to notify that it is complete.
-      synchronized (lock)
+      synchronized (sLock)
       {
         try
         {
-          lock.wait();
+          sLock.wait();
         }
         catch (Exception e)
         {
         }
       }
 
       if (yesNo)
-        result = alertResult;
+      {
+        result = AlertMessage.GetAlertResult();
+      }
     }
+    sIsAlertMessage = false;
+    sAlertMessageText = null;
     return result;
   }
 
+  public static String GetAlertMessageText()
+  {
+    return sAlertMessageText;
+  }
+
+  public static boolean IsAlertMessage()
+  {
+    return sIsAlertMessage;
+  }
+
+  public static void NotifyLock()
```

Comment:
Maybe it would make sense to specify what this lock is for, like `NotifyAlertMessageLock`. (Same with the actual variable name.)